tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,190 @@
+
+From the NServiceBus perspective, it makes no difference whether handlers are implemented inside one or multiple classes. When NServiceBus starts up, it will find all of these message handlers and automatically wire them up without any special configuration. Use it to group related handlers in order to make your code easier to understand.
+
+
+## Exercise
+
+Now let's take the solution we started in the last lesson and modify it to send a message. You can also grab the completed solution from the last lesson to use as a starting point.
+
+When we're done, the ClientUI endpoint will be sending a PlaceOrder message to itself, and then processing that message, as depicted in the following diagram:
+
+![Exercise 2 Diagram](diagram.png)
+
+
+### Create a messages assembly
+
+To share message between endpoints they need to be self-contained within a separate assembly, so let's create that assembly now.
+
+ 1. In the solution, create a new project and select the **Class Library** project type.
+ 1. Set the name of the project to **Messages**.
+ 1. Remove the automatically created **Class1.cs** file from the project. We would prefer to have our messages a little more organized than files in the root of the project. (More on this a little later.)
+ 1. Add the NServiceBus NuGet package to the Messages project.
+ 1. In the **ClientUI** project, add a reference to the **Messages** project.
+
+
+### Create a message
+
+In order to better organize messages, we'll create our first command in a folder called **Commands**.
+
+ 1. In the **Messages** project, create a new folder called **Commands**.
+ 1. In the **Commands** folder, add a new class named `PlaceOrder`.
+ 1. Mark `PlaceOrder` as `public` and implement `ICommand`.
+ 1. Add a public property of type `string` named `OrderId`.
+
+NOTE: The .NET Framework contains another interface named `ICommand` in the `System.Windows.Input` namespace. Be sure that if you use tooling to resolve the namespace, you select `NServiceBus.ICommand`. Most of the types you will need will reside in the `NServiceBus` namespace.
+
+When complete, your `PlaceOrder` class should look like the following:
+
+snippet:PlaceOrder
+
+
+### Create a handler
+
+Now that we've defined a message, we can create a corresponding message handler. For now, let's handle the message locally within the **ClientUI** endpoint.
+
+ 1. In the **ClientUI** project, create a new class named `PlaceOrderHandler`.
+ 1. Mark the handler class as public, and implement the `IHandleMessages<PlaceOrder>` interface.
+ 1. Add a logger instance, which will allow you to take advantage of the same logging system used by NServiceBus. This has an important advantage over `Console.WriteLine()`: the messages you write with the logger will be written to the log file in addition to the console. Use this code to add the logger instance to your handler class:
```

Comment:
"messages you write with the logger"  messages is an overloaded term. perhaps " entries written with the logger will be appear the log file in addition to the console. "